helm/prometheus-meta-operator/templates/prometheus-rules/inhibit.all.rules.yml

```diff
@@ -22,7 +22,7 @@ spec:
     - alert: InhibitionKubeStateMetricsDown
       annotations:
         description: '{{`KubeStateMetrics ({{ $labels.instance }}) is down.`}}'
-      expr: up{app="kube-state-metrics"} == 0 or absent(up{app="kube-state-metrics"}) == 1
+      expr: up{app="kube-state-metrics"} == 0 or absent(up{app="kube-state-metrics"} == 1)
```

Comment:
I fail to see how this fixes it. Could you explain why that works?

Reply:
`absent(up{app="kube-state-metrics"})` check if `up{app="kube-state-metrics"}` is missing, but it might be zero and present. While `absent(up{app="kube-state-metrics"} == 1)` check for its absence while value is 1. So it fires also when value is zero, and this aligns with `KubeStateMetricsMissing` [expression](https://github.com/giantswarm/prometheus-meta-operator/blob/32841dfa3a8a8a8d391667099657f7cf4454cd04/helm/prometheus-meta-operator/templates/prometheus-rules/up.all.rules.yml#L63)

Reply:
This inhibition is essentially a combination of the expression for the alert it needs to inhibit. 

```yaml
    - alert: KubeStateMetricsDown
      expr: up{app="kube-state-metrics"} == 0

    - alert: KubeStateMetricsMissing
      expr: absent(up{app="kube-state-metrics"} == 1)

    - alert: InhibitionKubeStateMetricsDown
      expr: up{app="kube-state-metrics"} == 0 or absent(up{app="kube-state-metrics"}) == 1
```

Reply:
Thanks Theo